### ISSUE TITLE
Coordinates not quantified / quantifiable

### ISSUE THREAD

**Issue by mueslo** [xarray-contrib/pint-xarray#124]
If I have a `DataArray` or `Dataset`, I would expect there to be some way to quantify not just the data, but also the coordinates. `<o>.pint.quantify()` seems to only quantify the data.

```python
import xarray as xr
import pint
import pint_xarray
ureg = pint.UnitRegistry(force_ndarray_like=True)

ds = xr.Dataset(
    {'z': (("x","y"), np.random.random((2, 3)), {"units": "m"})},
    coords={"x": ("x", [0, 1], {"units": "m"}), 
            "y": ("y", [5, 6, 7], {"units": "m"})})
ds = ds.pint.quantify(unit_registry=ureg)
ds.x + 2*ureg.cm
# DimensionalityError: Cannot convert from 'dimensionless' to 'centimeter'
```

Manually running quantify on the coordinate gives:

```python
ds.x.pint.quantify()
# ValueError: cannot create a Dataset from a DataArray with the same name as one of its coordinates
```

Manually constructing from quantified coordinates leads to an explicit dequantify:
```python
x = xr.DataArray([0, 1], attrs={'units': 'm'}, dims=('x',))
x = x.pint.quantify(unit_registry=ureg)

ds = xr.Dataset(
    {'z': (("x","y"), np.random.random((2, 3)), {"units": "m"})},
    coords={"x": x, 
            "y": ("y", [5, 6, 7], {"units": "m"})})
# UnitStrippedWarning: The unit of the quantity is stripped when downcasting to ndarray.
#ds = ds.pint.quantify(unit_registry=ureg)
ds.x + 2*ureg.cm
# DimensionalityError: Cannot convert from 'dimensionless' to 'centimeter'
```

Is this functionality not expected to work?

**Comment by mueslo** [comment]
Okay, it seems to be related to "anonymous" coordinates. What I was doing was more like

```python
ds = xr.Dataset(
    {'z': (("x","y"), np.random.random((2, 3)), {"units": "m"})},
    coords={"x": [0, 1], 
            "y": [5, 6, 7]})
ds.x.attrs['units'] = 'm' 
ds = ds.pint.quantify(unit_registry=ureg)
```
which leads to ds.z being correctly quantified, but bot ds.x

**Comment by mueslo** [comment]
https://pint-xarray.readthedocs.io/en/latest/generated/xarray.Dataset.pint.quantify.html#xarray-dataset-pint-quantify

**Comment by keewis** [comment]
Note that we're currently actively working on removing that restriction in `xarray` (will still need some time, though). Once that is done this should just work.

In the meantime, I guess we could make the note more obvious, maybe by not combining it with the one for `dask` or by moving the `dask` note to the `Notes` section and making this one a warning.

**Comment by keewis** [comment]
reopening, to be closed by #126
